src/Chromosome.java

```diff
@@ -1,9 +1,103 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
 public class Chromosome {
 
     public double[][] adjacencyMatrix;
 
-    public Chromosome(List<Double> dna) {}
+    public Chromosome(){}
 
+    /**
+     * Create layer from adjacency matrix
+     * @param adjacencyMatrix An adjacency matrix representing the network
+     */
+    public Chromosome(double[][] adjacencyMatrix) {
+        this.adjacencyMatrix = adjacencyMatrix;
+    }
+
+    /**
+     * Converts a Chromosome to a network
+     * @return Network made based off the chromosome
+     */
+    public Network ToNetwork(){
+        int numCols = adjacencyMatrix[0].length;
+        int numRows = adjacencyMatrix.length;
+        int priorLayerNode = -1;
+        Network net = new Network();
+        Layer inputLayer = new Layer(Type.INPUT);
+
+        List inputLayerIndices = IntStream.
+                range(0, numCols).
+                parallel().
+                filter((j) ->IntStream.range(0, numRows).parallel().allMatch(i -> adjacencyMatrix[i][j] != 0)).
+                mapToObj(obj -> obj).
+                collect(Collectors.toList());
+
+        // Adding nodes to layers must be done sequentially because the order is assumed to be maintained
+        // so to allow getting the indices to be done in parallel, list is collected and then nodes are added
+        inputLayerIndices.stream().
+                forEach(index -> addNodeToLayer((Integer) index, inputLayer));
```

Comment:
. on newline, in fact you can just have this one on a single line if you want.  Up to you, but if two liner than . on next line